### PR TITLE
Feat lp locking

### PR DESCRIPTION
**Pull request by subject026**

### CODE REVIEW

**Review by secbajor — COMMENTED** [review]
great! there is a lot here. I gave this a quick review and dropped a few quick comments, writing a bit more below as items that we can sync on when we have our next call. 

i checked this out locally and rebased it with the latest on dev/main -- there will be a few conflicts we have to make sure to resolve; mostly with the Bread/Power Icons 

I'm also noticing that currently with this PR i am seeing the link to lp token locking, so just noting that we need to check for the feature flag being enabled in at least one more place. 

i am seeing an opportunity to potentially refactor the useVotingPower context so that it could accept any wallet address, in this way we could use it to fetch both the logged in user voting power and a project's voting power, but this could happen later/separately.

File: src/app/governance/lp-vaults/page.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { notFound } from "next/navigation";
+import { Metadata } from "next";
+import { LPVotingPowerPage } from "./LPVotingPowerPage";
+import { parseFeatureVar } from "@/app/core/util/parseFeatureVar";
+
+export const metadata: Metadata = {
+  title: "Bread Governance",
```
[the remark this answers is not in the document]
Comment (subject026) [reply src/app/governance/lp-vaults/page.tsx:7]: Definitely. Best to ask Josh/Ruben what the title should be here I think.

[the remark this answers is not in the document]
Comment (diterra-code) [reply src/app/governance/lp-vaults/page.tsx:7]: Let us rename it to 'Bread voting'

[the remark this answers is not in the document]
Comment (subject026) [reply src/app/governance/lp-vaults/page.tsx:7]: This is for the LP vault page so 'Bread voting' might not be right for the page title

File: src/app/governance/context/VotingPowerContext.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      contracts: [
+        {
+          address: config.DISBURSER.address,
+          abi: DISTRIBUTOR_ABI,
+          functionName: "getVotingPowerForPeriod",
+          args: [
+            config.BREAD.address,
+            previousCycleStartingBlock,
+            lastClaimedBlocknumber,
+            user.address,
```
[the remark this answers is not in the document]
Comment (subject026) [reply src/app/governance/context/VotingPowerContext.tsx:109]: Agreed this is something we need to think about.

https://github.com/BreadchainCoop/crowdstaking-v2/issues/155

File: src/app/core/components/Icons/FistIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+export function FistIcon() {
```
[the remark this answers is not in the document]
Comment (subject026) [reply src/app/core/components/Icons/FistIcon.tsx:1]: Removed PowerIcon as the other one is being used more

File: src/app/governance/context/VotingPowerContext.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (subject026) [reply src/app/governance/context/VotingPowerContext.tsx]: I've split it into 2 context, one receives a wallet address.

File: src/app/bakery/components/Swap/Bake.tsx [hunk 4]
```diff
@@ -76,6 +76,7 @@ export default function Bake({
         return;
       }
       if (isSafe) {
+        // TODO look at using eth_getTransactionRecipt to catch submitted transactions
```
[the remark this answers is not in the document]
Comment (subject026) [reply src/app/bakery/components/Swap/Bake.tsx:79]: https://github.com/BreadchainCoop/crowdstaking-v2/issues/156

**Pull request merged**